src/Orleans.CodeGeneration/GrainReferenceGenerator.cs

```diff
@@ -285,14 +285,22 @@ private static ArgumentSyntax GetInvokeOptions(MethodInfo method)
             {
                 switch (option)
                 {
-                    case TransactionOption.NotSupported:
-                        options.Add(typeof(InvokeMethodOptions).GetNameSyntax().Member(InvokeMethodOptions.TransactionNotSupported.ToString()));
+                    case TransactionOption.Suppress:
                         break;
-                    case TransactionOption.Required:
-                        options.Add(typeof(InvokeMethodOptions).GetNameSyntax().Member(InvokeMethodOptions.TransactionRequired.ToString()));
+                    case TransactionOption.CreateOrJoin:
+                        options.Add(typeof(InvokeMethodOptions).GetNameSyntax().Member(InvokeMethodOptions.TransactionCreateOrJoin.ToString()));
                         break;
-                    case TransactionOption.RequiresNew:
-                        options.Add(typeof(InvokeMethodOptions).GetNameSyntax().Member(InvokeMethodOptions.TransactionRequiresNew.ToString()));
+                    case TransactionOption.Create:
+                        options.Add(typeof(InvokeMethodOptions).GetNameSyntax().Member(InvokeMethodOptions.TransactionCreate.ToString()));
+                        break;
+                    case TransactionOption.Mandatory:
+                        options.Add(typeof(InvokeMethodOptions).GetNameSyntax().Member(InvokeMethodOptions.TransactionMandatory.ToString()));
+                        break;
+                    case TransactionOption.Supported:
+                        options.Add(typeof(InvokeMethodOptions).GetNameSyntax().Member(InvokeMethodOptions.TransactionSupported.ToString()));
+                        break;
+                    case TransactionOption.Never:
```

Comment:
Perhaps `Never` should be `NotAllowed` or `Forbidden`. "Not allowed" is simple English, so I'm partial to that.

`Never` is odd, since it's a declaration that this method never has a transaction context when it's called, but it should instead indicate that it fails when there is a transaction context.

Reply:
(but I'm fine to leave this as-is)

Similarly, `Join` is consistent with the other names instead of `Mandatory`. Mandatory is ok, just not as consistent and not as obvious in terms of the resulting behavior.

Reply:
Meant to use Join instead of Mandatory, forgot to switch that over.  Will do.